saleor/dashboard/order/forms.py

```diff
@@ -181,8 +195,36 @@ def save(self):
             order.change_status('shipped')
 
 
+class CancelOrderForm(forms.Form):
+
+    def __init__(self, *args, **kwargs):
+        self.order = kwargs.pop('order')
+        super(CancelOrderForm, self).__init__(*args, **kwargs)
+
+    def clean(self):
+        data = super(CancelOrderForm, self).clean()
+        if not self.order.can_cancel():
+            raise forms.ValidationError(_('This order can\'t be cancelled'))
+        return data
+
+    def cancel_order(self):
+        self.order.change_status(status=Status.CANCELLED)
+
+
+class RemoveVoucherForm(forms.Form):
+
+    def __init__(self, *args, **kwargs):
+        self.order = kwargs.pop('order')
+        super(RemoveVoucherForm, self).__init__(*args, **kwargs)
+
+    def remove_voucher(self):
+        self.order.discount_amount = 0
+        self.order.discount_name = ''
+        self.order.voucher = None
```

Comment:
Please follow what stock handling does to avoid race conditions when allocating/deallocating.